<?xml version="1.0" encoding="UTF-8" ?>
<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom" xmlns:dc="http://purl.org/dc/elements/1.1/">
  <channel>
    <title>I have python code not working with python datasource[Error running query: &lt;class ‘NameError’&gt;]</title>
    <link>https://discuss.redash.io/t/i-have-python-code-not-working-with-python-datasource-error-running-query-class-nameerror/11594</link>
    <description>### Issue Summary

The following code does not work with python datasource and returns an error
```
def test(a):
    return a
print(test(1)) # No1
print(map(lambda v: test(v), [1,2,3])) # No2
print(list(map(lambda v: test(v), [1,2,3]))) # No3
print([test(v) for v in [1,2,3]]) # No4
```
No3, 4 will say Error running query: &lt;class &#39;NameError&#39;&gt; name &#39;test&#39; is not defined.

Running the same code from Redash OS works fine. (same python version)
Why does such a difference occur?

### Technical details:

* Redash Version:10.1.0
* Browser/OS:Chrome/Ubuntu16.04.6 LTS
* How did you install Redash:build docker-compose</description>
    
    <lastBuildDate>Fri, 17 Feb 2023 06:47:00 +0000</lastBuildDate>
    <category>Self Hosted Redash Support</category>
    <atom:link href="https://discuss.redash.io/t/i-have-python-code-not-working-with-python-datasource-error-running-query-class-nameerror/11594.rss" rel="self" type="application/rss+xml" />
      <item>
        <title>I have python code not working with python datasource[Error running query: &lt;class ‘NameError’&gt;]</title>
        <dc:creator><![CDATA[Hisasui]]></dc:creator>
        <description><![CDATA[
            <h3>
<a name="issue-summary-1" class="anchor" href="https://discuss.redash.io#issue-summary-1"></a>Issue Summary</h3>
<p>The following code does not work with python datasource and returns an error</p>
<pre><code class="lang-auto">def test(a):
    return a
print(test(1)) # No1
print(map(lambda v: test(v), [1,2,3])) # No2
print(list(map(lambda v: test(v), [1,2,3]))) # No3
print([test(v) for v in [1,2,3]]) # No4
</code></pre>
<p>No3, 4 will say Error running query: &lt;class ‘NameError’&gt; name ‘test’ is not defined.</p>
<p>Running the same code from Redash OS works fine. (same python version)<br>
Why does such a difference occur?</p>
<h3>
<a name="technical-details-2" class="anchor" href="https://discuss.redash.io#technical-details-2"></a>Technical details:</h3>
<ul>
<li>Redash Version:10.1.0</li>
<li>Browser/OS:Chrome/Ubuntu16.04.6 LTS</li>
<li>How did you install Redash:build docker-compose</li>
</ul>
          <p><a href="https://discuss.redash.io/t/i-have-python-code-not-working-with-python-datasource-error-running-query-class-nameerror/11594/1">Read full topic</a></p>
        ]]></description>
        <link>https://discuss.redash.io/t/i-have-python-code-not-working-with-python-datasource-error-running-query-class-nameerror/11594/1</link>
        <pubDate>Fri, 17 Feb 2023 06:47:00 +0000</pubDate>
        <guid isPermaLink="false">discuss.redash.io-post-11594-1</guid>
        <source url="https://discuss.redash.io/t/i-have-python-code-not-working-with-python-datasource-error-running-query-class-nameerror/11594.rss">I have python code not working with python datasource[Error running query: &lt;class ‘NameError’&gt;]</source>
      </item>
  </channel>
</rss>
